pkg/controller/secretsmanager/controller.go

```
@@ -0,0 +1,287 @@
+	}
+	if req.DeletedDate != nil {
+		resourceUpToDate = true
+		cr.Status.SetConditions(runtimev1.Deleting().WithMessage(secretMarkedForDeletion))
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:          true,
+		ResourceUpToDate:        resourceUpToDate,
+		ResourceLateInitialized: !cmp.Equal(current, &cr.Spec.ForProvider),
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.Secret)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotSecret)
+	}
+
+	cr.SetConditions(runtimev1.Creating())
+
+	secret, err := secretsmanager.GetSecret(ctx, e.kube, cr)
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+
+	_, err = e.client.CreateSecretRequest(secretsmanager.GenerateCreateSecretsmanagerInput(meta.GetExternalName(cr), &cr.Spec.ForProvider, secret)).Send(ctx)
+	return managed.ExternalCreation{}, errors.Wrap(err, errCreateFailed)
+}
+
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+
+	cr, ok := mg.(*v1alpha1.Secret)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotSecret)
+	}
+
+	req, err := e.client.DescribeSecretRequest(&awssecretsmanager.DescribeSecretInput{SecretId: awscommon.String(meta.GetExternalName(cr))}).Send(ctx)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(secretsmanager.IsErrorNotFound, err), errDescribeSecretFailed)
+	}
+
+	err = e.updateTags(ctx, cr, req)
+	if err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+
+	secret, err := secretsmanager.GetSecret(ctx, e.kube, cr)
+	if err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+
+	_, err = e.client.UpdateSecretRequest(secretsmanager.GenerateUpdateSecretInput(meta.GetExternalName(cr), cr.Spec.ForProvider, secret)).Send(ctx)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(secretsmanager.IsErrorNotFound, err), errUpdateFailed)
```

Why ignore this error?